reducers/shoppingReducer.js

```diff
@@ -0,0 +1,102 @@
+import {
+  FETCH_BANNERS,
+  FETCH_CATEGORIES,
+  FETCH_PRODUCTS,
+  SET_CATEGORY,
+  SET_CART_OPEN,
+  ADD_TO_SHIPPING,
+  DELETE_FROM_SHIPPING,
+  RESET_STORE,
+  SIGN_IN,
+  SIGN_OUT,
+} from "../actions/types";
+import _ from "lodash";
+
+const shoppingReducer = (state = {}, action) => {
+  switch (action.type) {
+    case FETCH_BANNERS:
+      return { ...state, banners: action.payload };
+    case FETCH_CATEGORIES:
+      return { ...state, categories: action.payload };
+    case FETCH_PRODUCTS:
+      return { ...state, products: action.payload };
+    case SET_CATEGORY:
+      return { ...state, selected: action.payload };
+    case SET_CART_OPEN:
+      return { ...state, cartOpen: action.payload };
+    case ADD_TO_SHIPPING:
+      let { cart, totalPrice } = addToCart(
+        action.payload,
+        state.cartItems?.cart,
+        state.cartItems?.totalPrice
+      );
+      return { ...state, cartItems: { ...state.cartItems, cart, totalPrice } };
+    case DELETE_FROM_SHIPPING:
+      let { deleteCart, newTotalPrice } = deleteFromCart(
+        action.payload,
+        state.cartItems?.cart,
+        state.cartItems?.totalPrice
+      );
+      return {
+        ...state,
+        cartItems: {
+          ...state.cartItems,
+          cart: deleteCart,
+          totalPrice: newTotalPrice,
+        },
+      };
+    case SIGN_IN:
+      localStorage.setItem("user", action.payload.email);
+      return { ...state, userId: action.payload.email };
+
+    case SIGN_OUT:
+      localStorage.clear();
+      return { ...state, userId: null };
+    case RESET_STORE: {
+      state = {};
+      return state;
+    }
+    default:
+      return state;
+  }
+};
+
+const addNewProduct = (product, cart, totalPrice = 0) => {
```

Comment:
Can we create a util file to have these functions?

Reply:
done